ktlint-ruleset-template/build.gradle.kts

```diff
@@ -0,0 +1,71 @@
+plugins {
+    `ktlint-kotlin-common` // replace it with 'org.jetbrains.kotlin.jvm'
+    `java-library`
+    `maven-publish`
+}
+
+group = "com.github.username"
+
+val sourcesJar by tasks.registering(Jar::class) {
+    dependsOn(tasks.classes)
+    archiveClassifier.set("sources")
+    from(sourceSets.main.get().allSource)
+}
+
+val javadocJar by tasks.registering(Jar::class) {
+    dependsOn(tasks.javadoc)
+    archiveClassifier.set("javadoc")
+    from(tasks.javadoc.get().destinationDir)
+}
+
+artifacts {
+    archives(sourcesJar)
+    archives(javadocJar)
+}
+
+val ktlint: Configuration = configurations.create("ktlint")
+
+dependencies {
+    ktlint(projects.ktlint)
+
+    compileOnly(projects.ktlintCore)
+
+    testImplementation(projects.ktlintCore)
+    testImplementation(projects.ktlintTest)
+    testImplementation(libs.junit5)
+    testImplementation(libs.assertj)
+}
+
+tasks.test {
+    useJUnitPlatform()
+}
+
+tasks.register<JavaExec>("ktlint") {
+    dependsOn(tasks.classes)
+    group = LifecycleBasePlugin.VERIFICATION_GROUP
+    mainClass.set("com.pinterest.ktlint.Main")
+    // adding compiled classes to the classpath so that ktlint would validate project"s sources
+    // using its own ruleset (in other words to dogfood)
+    classpath = ktlint + sourceSets.main.get().output
+    args("--debug", "src/**/*.kt")
+}.let {
+    tasks.check.get().dependsOn(it)
+}
```

Comment:
The `ktlint` task's configuration in `ktlint-ruleset-template` is not same with `ktlint` task in root `build.gradle`, I'm not sure which one is right.

https://github.com/pinterest/ktlint/blob/c42647182196339f08ff14cea9dd0321d69014fc/build.gradle#L30-L39

Reply:
The main diffs are `dependsOn`.

Reply:
The root `build.gradle` is definitively the correct one. When you run it, you can see that it parses about 220+ files. I expect that the version in `ktlint-ruleset-template` is meant as example for developers who want to write their own custom ruleset.

Reply:
Ok.